src/modules/job-manager/Makefile.am

```diff
@@ -12,6 +12,12 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/src/common/libccan \
 	$(JANSSON_CFLAGS)
 
+COMMON_LIBS = \
+	job-manager.la \
```

Comment:
was this supposed to be `libjob-manager.la`?

The asan build is failing with 
```
 *** Warning: Linking the executable test_job.t against the loadable module
  *** job-manager.so is not portable!
  /usr/bin/ld: test/job_t-job.o: in function `test_create':
  /usr/src/src/modules/job-manager/test/job.c:25: undefined reference to `job_create'
  /usr/bin/ld: /usr/src/src/modules/job-manager/test/job.c:28: undefined reference to `ok_at_loc'
```

Looks like somehow the link to libtap for the tests was dropped as well?

Reply:
This was intentionally `job-manager.la` so that the modules would like to the dynamic library.  If I'm reading this right, `libjob-manager.la` is the static library that gets linked into `job-manager.la`, and the linker script exposes all the symbols the modules need so this seemed safer to avoid duplication.

Reply:
Yes, I can see the sense in that, but the test programs specifically should be linked against the `libjob-manager.la` since they are just unit tests.

Also, I've never tried to link one plugin DSO against another so I'd be surprised if there weren't issues there. I'm still wondering what symbols internal to the job-manager anything besides the unit tests are using? (Since that breaks the separation between the job-manager and its jobtap plugins, I should probably fix that.)

Reply:
I think you could probably just revert the changes to `test_ldadd`. The unit tests are purposefully built to pull in code from the current directory and avoid dynamic linking as much as practical. Were there any specific issues with the unit tests on nixos? Seems like the issue was instead with the 3 external job-manager plugins.

Reply:
> I can definitely use the static lib in the tests, but to clarify you want the static lib there to embed the symbols in the binary rather than using the SO like we use in production?

I admit I don't really understand the issue you are having (a bit above my level of understand of dynamic loading), so if you say the job-manager/plugins need to be linked against the `job-manager.so` I trust you. However, the warning is a bit concerning (and annoying since it comes up in every build). Is there really no other way? What symbols were missing on dlopen()?

> I can definitely use the static lib in the tests, but to clarify you want the static lib there to embed the symbols in the binary rather than using the SO like we use in production?

Yes, the tests are unit tests so they are specifically testing the internal API of the individual source files.

Reply:
Do you have a recipe to reproduce the issue on NixOS?

I just remembered we have some other job-manager plugins used in in test under `t/job-manager/plugins` -- any sense of why those do not have the same issue exhibited by the plugins that are under `src/modules/job-manager/plugins`?

Reply:
I can use libjob-manager for all of these and it will work, that gets rid of the warning because they're no longer linking to an so with a non-canonical name.  That means we may end up with some amount of binary bloat, but the loader will de-duplicate at runtime and everything will continue to work.  Here, I'll push that version so we can see what the CI output looks like.